python/ray/tests/test_tf_differentiable.py

```diff
@@ -0,0 +1,623 @@
+from functools import reduce
+
+import pytest
+import tensorflow as tf
+import numpy as np
+
+import ray
+from ray.experimental.tf_utils import tf_differentiable
+
+@pytest.fixture
+def start_ray():
+    ray.init()
+
+    yield None
+
+    # teardown
+    ray.shutdown()
+
+@pytest.fixture
+def ray_actor():
+    @ray.remote
+    class Actor(object):
+        """ Ray Actor with TF differentiable functions. """
+
+        def __init__(self):
+            # enable eager execution
+            tf.enable_eager_execution()
+
+        # single input, single output
+        @tf_differentiable(num_return_vals=1)
```

Comment:
`num_return_vals` should be specified in `@ray.method(num_return_vals=...)`, not in `tf_differentiable`.

Also, 1 is the default, it should only be specified for values != 1